pkg/console/status/auth_status.go

```diff
@@ -0,0 +1,202 @@
+		componentNamespace: componentNamespace,
+		fieldManager:       fieldManager,
+		conditionsToApply:  map[AuthenticationConditionType]*metav1.Condition{},
+	}
+}
+
+// Degraded sets the Degraded condition to True and Progressing to False
+func (c *AuthStatusHandler) Degraded(reason, message string) *AuthStatusHandler {
+	now := metav1.Now()
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionTrue, reason, message, now)
+
+	return c
+}
+
+// Progressing sets the Progressing condition to True and Degraded to False
+func (c *AuthStatusHandler) Progressing(reason, message string) *AuthStatusHandler {
+	now := metav1.Now()
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionTrue, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionFalse, reason, message, now)
+
+	return c
+}
+
+// Unavailable sets all conditions to False
+func (c *AuthStatusHandler) Unavailable(reason, message string) *AuthStatusHandler {
+	now := metav1.Now()
+	c.setCondition(ConditionTypeAvailable, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionFalse, reason, message, now)
+
+	return c
+}
+
+// Available sets the Available condition to True, and Progressing and Degraded to False
+func (c *AuthStatusHandler) Available(reason, message string) *AuthStatusHandler {
+	now := metav1.Now()
+	c.setCondition(ConditionTypeAvailable, metav1.ConditionTrue, reason, message, now)
+	c.setCondition(ConditionTypeProgressing, metav1.ConditionFalse, reason, message, now)
+	c.setCondition(ConditionTypeDegraded, metav1.ConditionFalse, reason, message, now)
+
+	return c
+}
+
+func (c *AuthStatusHandler) setCondition(conditionType AuthenticationConditionType, status metav1.ConditionStatus, reason, message string, ts metav1.Time) *AuthStatusHandler {
+	if c.conditionsToApply[conditionType] == nil {
+		c.conditionsToApply[conditionType] = &metav1.Condition{Type: string(conditionType)}
+	}
+
+	c.conditionsToApply[conditionType].Status = status
+	c.conditionsToApply[conditionType].Reason = reason
+	c.conditionsToApply[conditionType].Message = message
+	c.conditionsToApply[conditionType].LastTransitionTime = ts
+
+	return c
+}
+
+func (c *AuthStatusHandler) WithCurrentOIDCClient(currentClientID string) *AuthStatusHandler {
+	c.currentClientID = currentClientID
+	return c
+}
+
+func (c *AuthStatusHandler) Apply(ctx context.Context, authnConfig *configv1.Authentication) error {
+	if authnConfig.Spec.Type != configv1.AuthenticationTypeOIDC {
+		return fmt.Errorf("authentication type not OIDC")
+	}
+
+	applyConfig, err := configv1ac.ExtractAuthenticationStatus(authnConfig, c.fieldManager)
+	if err != nil {
+		return err
+	}
+
+	if applyConfig.Status == nil {
+		applyConfig.Status = &configv1ac.AuthenticationStatusApplyConfiguration{}
+	}
+
+	idx := -1
```

Comment:
you should name this variable properly, you're using it too often for it to have a generic name

Reply:
dropped it because of this: https://github.com/openshift/console-operator/pull/811#discussion_r1427717874